corehq/apps/app_manager/tasks.py

```diff
@@ -1,9 +1,23 @@
 from celery.task import task
 from corehq.apps.users.models import CommCareUser
+from corehq.util.decorators import serial_task
 
 
 @task(queue='background_queue', ignore_result=True)
 def create_user_cases(domain_name):
     from corehq.apps.callcenter.utils import sync_usercase
     for user in CommCareUser.by_domain(domain_name):
         sync_usercase(user)
+
+
+@serial_task('{app._id}', max_retries=0)
+def make_async_build(app):
+    latest_build = app.get_latest_app(released_only=False)
+    if latest_build.version == app.version:
+        return
+    errors = app.validate_app()
+    if not errors:
```

Comment:
what kind of ui are you imagining? since this is triggered by the phone, the logical place to alert the user would be on the phone but im not sure we have that capability. It is tough to alert on hq since they could be on any page (or not on hq at all) when this build is triggered, and will not even be logged in as the same user (mobile user on the phone, web user on hq). If the build fails it will likely cause problems on mobile anyway, that should become apparent.
In general, i like the idea of alerting but i'm not really sure how or what it would like in this case.